page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/DefaultPage.java

```diff
@@ -0,0 +1,168 @@
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.PageException if page was not
+     * initialized, or required element couldn't be found
+     */
+    @ActionTitles({
+            @ActionTitle("ru.sbtqa.tag.pagefactory.click.link"),
+            @ActionTitle("ru.sbtqa.tag.pagefactory.click.button")})
+    public void click(String elementTitle) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.click(element);
+    }
+
+    /**
+     * Press specified key on a keyboard Add corresponding parameter to allure
+     * report
+     *
+     * @param keyName name of the key. See available key names in {@link Keys}
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.press.key")
+    public void pressKey(String keyName) {
+        pageActions.press(null, keyName);
+    }
+
+    /**
+     * Focus a WebElement, and send specified key into it Add corresponding
+     * parameter to allure report
+     *
+     * @param keyName name of the key. See available key names in {@link Keys}
+     * @param elementTitle title of WebElement that accepts key commands
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.ElementNotFoundException if
+     * couldn't find element with required title
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.press.key")
+    public void pressKey(String keyName, String elementTitle) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.press(element, keyName);
+    }
+
+    /**
+     * TODO
+     *
+     * @param elementTitle WebElement that is supposed to be selectable
+     * @param option option to select
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.PageException if required
+     * element couldn't be found, or current page isn't initialized
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.select")
+    public void select(String elementTitle, String option) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.select(element, option);
+    }
+
+    /**
+     * Find web element with corresponding title, if it is a check box, select
+     * it If it's a WebElement instance, check whether it is already selected,
+     * and click if it's not Add corresponding parameter to allure report
+     *
+     * @param elementTitle WebElement that is supposed to represent checkbox
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.PageException if page was not
+     * initialized, or required element couldn't be found
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.select.checkbox")
+    public void setCheckBox(String elementTitle) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.setCheckbox(element, true);
+    }
+
+
+    /**
+     * TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)